src/main/java/org/ow2/proactive/catalog/service/BucketService.java

```diff
@@ -188,18 +196,28 @@ public BucketMetadata getBucketMetadata(String bucketName) {
             Optional<String> objectName) {
         List<BucketMetadata> entities;
         List<String> owners = Collections.singletonList(ownerName);
-
+        List<String> kindList = new ArrayList<>();
+        if (kind.isPresent()) {
+            kindList = Arrays.asList(kind.get().toLowerCase().split(","));
+        }
         if (!StringUtils.isEmpty(ownerName)) {
             entities = getBucketEntities(owners, kind, contentType, objectName);
         } else if (kind.isPresent() || contentType.isPresent() || objectName.isPresent()) {
             List<Object[]> bucketsFromDB;
             if (contentType.isPresent() || objectName.isPresent()) {
-                bucketsFromDB = bucketRepository.findContainingKindAndContentTypeAndObjectName(kind.orElse(""),
-                                                                                               contentType.orElse(""),
-                                                                                               objectName.orElse(""),
-                                                                                               sortById);
+                if (kind.isPresent()) {
+                    bucketsFromDB = bucketRepository.findContainingKindAndContentTypeAndObjectName(kindList,
+                                                                                                   contentType.orElse(""),
+                                                                                                   objectName.orElse(""),
+                                                                                                   sortById);
+                } else {
+                    bucketsFromDB = bucketRepository.findContainingContentTypeAndObjectName(contentType.orElse(""),
+                                                                                            objectName.orElse(""),
+                                                                                            sortById);
+
+                }
             } else { // only kind.isPresent()
-                bucketsFromDB = bucketRepository.findContainingKind(kind.orElse(""), sortById);
+                bucketsFromDB = bucketRepository.findContainingKind(kindList, sortById);
```

Comment:
what happens when the list is empty ? no default ?

Reply:
kindList cannot be empty as it is used when kind.isPresent() is true